modules/FindFreetype.cmake

```diff
@@ -159,6 +185,59 @@ find_package_handle_standard_args(
     FREETYPE_VERSION_STRING
 )
 
+if(FREETYPE_FOUND)
+  # Determine if Freetype was built as static or shared
+  set(FREETYPE_SHARED OFF)
+  foreach(_path ${FREETYPE_LIBRARIES})
+    string(FIND ${_path} "/" _result REVERSE)
+    string(SUBSTRING ${_path} ${_result} -1 _name)
+    foreach(_hint dll so dylib)
+      string(FIND ${_name} ${_hint} _result)
+      if(NOT _result EQUAL -1)
+        set(FREETYPE_SHARED ON)
+        break()
+      endif()
+    endforeach()
+  endforeach()
+
+  # If Freetype was built static, we need to link its optional dependencies
+  if(NOT FREETYPE_SHARED)
+    # We can read which dependencies Freetype was built with from ftoption.h
+    file(READ "${FREETYPE_INCLUDE_DIR_freetype2}/freetype/config/ftoption.h" _ftoption_h)
+    foreach(_dep ZLIB BZip2 PNG HarfBuzz)
+      string(TOUPPER ${_dep} _DEP)
+      set(_require_dep OFF)
+
+      # Use regex to check if a depdency is needed
+      # Example of when a dependency was used to build Freetype:
+      # #define FT_CONFIG_OPTION_USE_BZIP2
+      # Example of when a dependency was not used to build Freetype:
+      # /* #define FT_CONFIG_OPTION_USE_HARFBUZZ */
+      # We want to match the first and not the second
+      set(_match_before "\n[ \t]*\\#define[ \t]*FT_CONFIG_OPTION")
+      set(_match_after "[ \t]*\n")
+      string(REGEX MATCH "${_match_before}_USE_${_DEP}${_match_after}" _use_dep "${_ftoption_h}")
```

Comment:
I would move this line to the `else()` clause so the handling of `_USE_` and `_SYSTEM_` looks symmetric.